vlog4j-core/src/test/java/org/semanticweb/vlog4j/core/reasoner/FileDataSourceUtils.java

```diff
@@ -68,4 +84,60 @@ private FileDataSourceUtils() {
 		return content;
 	}
 
+	public static void testConstructor(final FileDataSource unzippedFileDataSource,
+			final FileDataSource zippedFileDataSource)
+					throws IOException {
+		final String expectedDirCanonicalPath = new File(INPUT_FOLDER).getCanonicalPath();
+		final File unzippedFile = unzippedFileDataSource.getFile();
+		final File zippedFile = zippedFileDataSource.getFile();
+
+		assertEquals(unzippedFile, unzippedFileDataSource.getFile());
+		assertEquals(expectedDirCanonicalPath, unzippedFileDataSource.getDirCanonicalPath());
```

Comment:
I see what you mean. There is, however, a huge problem: What would those expected parameters look like? At the moment, the parameters that the function takes are abstract `FileDataSource` objects and not paths. One of the major ideas when writing these tests was to create the concrete `CsvFileDataSource` or `RdfFileDataSource` objects outside of these functions and to only work with the generalised `FileDataSource` in the utility class. If I passed paths to the utility class, I would have to pass the concrete class as parameter as well and create the concrete objects with a generic constructor, which is very troublesome and ugly in Java. Moreover, as far as I understand, constructors with parameters are not supported in that approach, which is not compatible with `CsvFileDataSource` or `RdfFileDataSource`, because they expect a path.